components/calendar/src/any_calendar.rs

```diff
@@ -1073,33 +1073,24 @@ impl AnyCalendarKind {
             .and_then(Self::get_for_bcp47_value)
     }
 
-    /// Extract the calendar component from a [`DataLocale`]
-    ///
-    /// Returns `None` if the calendar is not specified or unknown. If you prefer an error, use
-    /// [`CalendarError::unknown_any_calendar_kind`].
-    fn get_for_data_locale(l: &DataLocale) -> Option<Self> {
-        l.get_unicode_ext(&key!("ca"))
-            .and_then(|v| Self::get_for_bcp47_value(&v))
-    }
-
     // Do not make public, this will eventually need fallback
     // data from the provider
     fn from_data_locale_with_fallback(l: &DataLocale) -> Self {
-        if let Some(kind) = Self::get_for_data_locale(l) {
-            kind
-        } else {
-            let lang = l.language();
-            if lang == language!("th") {
-                Self::Buddhist
-            // Other known fallback routes for currently-unsupported calendars
-            // } else if lang == language!("sa") {
-            //     Self::IslamicUmalqura
-            // } else if lang == language!("af") || lang == language!("ir") {
-            //     Self::Persian
-            } else {
-                Self::Gregorian
-            }
-        }
+        l.keywords
+            .get(&key!("ca"))
+            .and_then(Self::get_for_bcp47_value)
+            .unwrap_or_else(|| {
+                if l.id.language == language!("th") {
+                    Self::Buddhist
+                // Other known fallback routes for currently-unsupported calendars
+                // } else if lang == language!("sa") {
+                //     Self::IslamicUmalqura
+                // } else if lang == language!("af") || lang == language!("ir") {
+                //     Self::Persian
```

Comment:
Observation: These calendars have been supported since 1.4 but this seems to have gotten lost at some point; we should make sure there's an issue tracking it